pkg/components/headlamp/component.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package headlamp
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+
+	"github.com/kinvolk/lokomotive/internal/template"
+	"github.com/kinvolk/lokomotive/pkg/components"
+	"github.com/kinvolk/lokomotive/pkg/components/util"
+	"github.com/kinvolk/lokomotive/pkg/k8sutil"
+)
+
+const name = "headlamp"
+
+const chartValuesTmpl = `
+nodeSelector:
+  beta.kubernetes.io/os: linux
+image:
+  # TODO use v0.1.0 when it's released
+  tag: l8e
+  pullPolicy: Always
+ingress:
+  enabled: true
+  hosts:
+  - host: {{ .IngressHost }}
+    paths:
+    - /
+  tls:
+  - secretName: {{ .IngressHost }}-tls
+    hosts:
+    - {{ .IngressHost }}
+  annotations:
+    cert-manager.io/cluster-issuer: {{ .CertManagerClusterIssuer }}
+    contour.heptio.com/websocket-routes: "/"
+`
+
+//nolint:gochecknoinits
+func init() {
+	components.Register(name, newComponent())
+}
+
+type component struct {
+	IngressHost              string `hcl:"ingress_host,attr"`
```

Comment:
I think we should use `pkg/components/types.Ingress` here for both properties, so we are consistent with other components.

Reply:
Good point.